rfcs/0003-buildpack-yml-to-env-var.md

```diff
@@ -0,0 +1,90 @@
+# Buildpack.yml to Environment Variables
+
+## Proposal
+
+Migrate to using environment variables to do all buildpack configuration and
+get rid of `buildpack.yml`.
+
+## Motivation
+
+There are several reasons for making this switch.
+1. There is already an existing RFC that proposes moving away from
+   `buildpack.yml` as a configuration tool.
+1. Environment variables appears to be the standard for configuration in other
+   buildpack ecosystems such as Google Buildpacks and Heroku as well as the
+   Paketo Java buildpacks. Making this change will align the buildpack with the
+   rest of the buildpack ecosystem.
+1. There is native support to pass environment variables to the buildpack
+   either on a per run basis or by configuration that can be checked into
+   source control, in the form of `project.toml`.
+1. Go and Dotnet-core buildpacks are moving to this strategy.
+
+## Implementation
+The proposed environment variables are as follows:
+
+### Existing configurations
+
+#### BP_NODE_VERSION
+```shell
+$BP_NODE_VERSION="~10"
+```
+This will replace the following structure in `buildpack.yml`:
+```yaml
+nodejs:
+  version: ~10
+```
+
+#### BP_NODE_OPTIMIZE_MEMORY
+```shell
+$BP_NODE_OPTIMIZE_MEMORY=true
+```
+This will replace the following structure in `buildpack.yml`:
+```yaml
+nodejs:
+  optimize-memory: true
+```
+
+### Proposed additions
+
+#### BP_NODE_PROJECT_PATH
+```shell
+$BP_NODE_PROJECT_PATH=./src/nodejs
+```
+If set, this relative path will be considered as the root of the app to be
+built by nodejs buildpacks.
+
+There is no current `buildpack.yml` support for this.
+
+#### BP_NODE_START_ENTRYPOINT
```

Comment:
@arjun024 it feels like we could have a generic entrypoint environment variable that applies to all buildpacks, rather than separate entrypoint env. vars. Something like `BP_ENTRYPOINT` or `PAKETO_ENTRYPOINT` (similar to what Google buildpacks do)

Reply:
I also think a setting like `BP_ENTRYPOINT` makes sense.

Reply:
Perhaps we should move `BP_ENTRYPOINT` to a separate project-wide RFC, as it would apply to all buildpacks?

Reply:
@kvedurmu Sure, this defines the behavior of the env var in the node context

Reply:
Alternative choices for env names: `BP_LAUNCHPOINT`, `BP_STARTPOINT`.
@ekcasey @kvedurmu @ryanmoran @ForestEckhardt Suggestions?